pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2883,10 +2883,9 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
           // 2. Proactively delete the oldest data snapshot to make sure that we only keep at most 2 data snapshots
           //    at any time in case of REFRESH use case.
           if (forceCleanup) {
-            if (lineageEntry.getState() == LineageEntryState.IN_PROGRESS && !Collections
-                .disjoint(segmentsFrom, lineageEntry.getSegmentsFrom())) {
```

Comment:
hey folks, I accidentally pushed my change of using disjoint() in https://github.com/apache/pinot/pull/8240/files. Tests seemed fine with that more strict check. 

But I'm reverting it back here to keep this PR focused. If that's actually a right move, I'll add it back in following PR with more proper UT cases.